### PR TITLE
Fix memory usage for most java tools

### PR DESCRIPTION
**Pull request by matthdsm**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/rtgtools/format/main.nf [hunk 1]
```diff
@@ -29,7 +29,7 @@ process RTGTOOLS_FORMAT {
     if (!task.memory) {
         log.info '[RTG format] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
     } else {
-        avail_mem = task.memory.giga + "G"
+        avail_mem = (task.memory.mega*0.8).intValue() + "G"
```
Comment (matthdsm) [inline modules/nf-core/rtgtools/format/main.nf:32]:
```suggestion
        avail_mem = (task.memory.mega*0.8).intValue() + "M"
```

File: modules/nf-core/rtgtools/format/main.nf [hunk 2]
```diff
@@ -53,7 +53,7 @@ process RTGTOOLS_FORMAT {
...
```
Comment (pontus) [inline modules/nf-core/rtgtools/format/main.nf:56]: G -> M

File: modules/nf-core/rtgtools/format/main.nf (hunk 1 above)
Comment (pontus) [inline modules/nf-core/rtgtools/format/main.nf]: G -> M

File: modules/nf-core/picard/liftovervcf/main.nf [hunk 3]
```diff
@@ -28,11 +28,11 @@ process PICARD_LIFTOVERVCF {
     if (!task.memory) {
```
Comment (pontus) [inline modules/nf-core/picard/liftovervcf/main.nf:28]: Default needs to change here (1 - > 1024).

File: modules/nf-core/rtgtools/format/main.nf (hunk 2 above)
Comment (matthdsm) [inline modules/nf-core/rtgtools/format/main.nf:56]:
```suggestion
        avail_mem = (task.memory.mega*0.8).intValue() + "M"
```

File: modules/nf-core/picard/fastqtosam/main.nf [hunk 4]
```diff
@@ -28,7 +28,7 @@ process PICARD_FASTQTOSAM {
     def sample_name = args.contains("--SAMPLE_NAME") || args.contains("-SM") ? "" : "--SAMPLE_NAME ${prefix}"
```
Comment (pontus) [inline modules/nf-core/picard/fastqtosam/main.nf:28]: Entire fix needed.

File: modules/nf-core/malt/build/main.nf [hunk 5]
```diff
@@ -26,7 +26,7 @@ process MALT_BUILD {
     if (!task.memory) {
```
Comment (pontus) [inline modules/nf-core/malt/build/main.nf:26]: Default needs change (6 -> 6144).

File: modules/nf-core/gatk4/variantfiltration/main.nf [hunk 6]
```diff
@@ -25,14 +25,14 @@ process GATK4_VARIANTFILTRATION {
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
 
-    def avail_mem = 3
+    def avail_mem = 3072
     if (!task.memory) {
         log.info '[GATK VariantFiltration] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
     } else {
         avail_mem = task.memory.toGiga()
```
Comment (pontus) [inline modules/nf-core/gatk4/variantfiltration/main.nf:32]:
```suggestion
        avail_mem = (task.memory.mega*0.8).intValue()
```

**Review by pontus — APPROVED** [review]
LGTM

File: modules/nf-core/gatk4/leftalignandtrimvariants/main.nf [hunk 7]
```diff
@@ -25,14 +25,14 @@ process GATK4_LEFTALIGNANDTRIMVARIANTS {
     def args             = task.ext.args   ?: ''
     def prefix           = task.ext.prefix ?: "${meta.id}"
     def interval_command = intervals       ? "--intervals $intervals" : ""
-    def avail_mem = 3
+    def avail_mem = 3072
     if (!task.memory) {
         log.info '[GATK LeftAlignAndTrimVariants] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
     } else {
         avail_mem = task.memory.toGiga()
```
Comment (drpatelh) [inline modules/nf-core/gatk4/leftalignandtrimvariants/main.nf:32]: Need to update this?

File: modules/nf-core/gatk4/leftalignandtrimvariants/main.nf [hunk 8]
```diff
@@ -25,14 +25,14 @@ process GATK4_LEFTALIGNANDTRIMVARIANTS {
...
     }
     """
-    gatk --java-options "-Xmx${avail_mem}G" LeftAlignAndTrimVariants \\
+    gatk --java-options "-Xmx${avail_mem}M" LeftAlignAndTrimVariants \\
```
Comment (drpatelh) [inline modules/nf-core/gatk4/leftalignandtrimvariants/main.nf:35]:
```suggestion
    gatk --java-options "-Xmx${avail_mem}M" \\
        LeftAlignAndTrimVariants \\
```

File: modules/nf-core/gatk4/leftalignandtrimvariants/main.nf (hunk 7 above)
Comment (matthdsm) [reply modules/nf-core/gatk4/leftalignandtrimvariants/main.nf]: 👍🏻 Done!

File: modules/nf-core/picard/liftovervcf/main.nf [hunk 9]
```diff
@@ -24,15 +24,15 @@ process PICARD_LIFTOVERVCF {
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def avail_mem = 1
+    def avail_mem = 1024
```
Comment (drpatelh) [inline modules/nf-core/picard/liftovervcf/main.nf:27]:
```suggestion
    def avail_mem = 3072
```

File: modules/nf-core/snpeff/download/main.nf [hunk 10]
```diff
@@ -23,11 +23,11 @@ process SNPEFF_DOWNLOAD {
     if (!task.memory) {
         log.info '[snpEff] Available memory not known - defaulting to 6GB. Specify process memory requirements to change this.'
     } else {
-        avail_mem = task.memory.giga
+        avail_mem = (task.memory.mega*0.8).intValue()
```
Comment (drpatelh) [inline modules/nf-core/snpeff/download/main.nf:26]: Can't comment on that line but need to change the `def avail_mem = 6` to megabytes

File: modules/nf-core/snpeff/snpeff/main.nf [hunk 11]
```diff
@@ -28,13 +28,13 @@ process SNPEFF_SNPEFF {
     if (!task.memory) {
         log.info '[snpEff] Available memory not known - defaulting to 6GB. Specify process memory requirements to change this.'
     } else {
-        avail_mem = task.memory.giga
+        avail_mem = (task.memory.mega*0.8).intValue()
```
Comment (drpatelh) [inline modules/nf-core/snpeff/snpeff/main.nf:31]: Can't comment on that line but need to change the `def avail_mem = 6` to megabytes

File: modules/nf-core/trinity/main.nf [hunk 12]
```diff
@@ -35,15 +35,15 @@ process TRINITY {
     if (!task.memory) {
         log.info '[Trinity] Available memory not known - defaulting to 7GB. Specify process memory requirements to change this.'
     } else {
-        avail_mem = task.memory.giga
+        avail_mem = (task.memory.mega*0.8).intValue()
```
Comment (drpatelh) [inline modules/nf-core/trinity/main.nf:38]: Can't comment on that line but need to change the `def avail_mem = 7` to megabytes

**Review by drpatelh — COMMENTED** [review]
Thanks @matthdsm ! Few more updates required.

File: modules/nf-core/gatk4/leftalignandtrimvariants/main.nf (hunk 7 above)
Comment (drpatelh) [reply modules/nf-core/gatk4/leftalignandtrimvariants/main.nf]: Sorry, I meant changing the line `avail_mem = task.memory.toGiga()` to the new 0.8 calculation.

File: modules/nf-core/glnexus/main.nf [hunk 13]
```diff
@@ -23,11 +23,11 @@ process GLNEXUS {
 
     // Make list of GVCFs to merge
     def input = gvcfs.collect { it.toString() }
-    def avail_mem = 3
+    def avail_mem = 3072
     if (!task.memory) {
         log.info '[Glnexus] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
     } else {
-        avail_mem = task.memory.giga
+        avail_mem = (task.memory.mega*0.8).intValue()
```
[the remark this answers is not in the document]
Comment (matthdsm) [reply modules/nf-core/glnexus/main.nf:30]: No idea. I'm not familiar with the tool, so I'll revert it

File: modules/nf-core/gatk4/leftalignandtrimvariants/main.nf (hunk 8 above)
Comment (matthdsm) [reply modules/nf-core/gatk4/leftalignandtrimvariants/main.nf]: Fixed!

File: modules/nf-core/glnexus/main.nf (hunk 13 above)
[the remark this answers is not in the document]
Comment (matthdsm) [reply modules/nf-core/glnexus/main.nf]: Fixed!

File: modules/nf-core/picard/liftovervcf/main.nf (hunk 9 above)
Comment (matthdsm) [reply modules/nf-core/picard/liftovervcf/main.nf]: Fixed

File: modules/nf-core/snpeff/snpeff/main.nf (hunk 11 above)
Comment (matthdsm) [reply modules/nf-core/snpeff/snpeff/main.nf:31]: Fixed!

**Comment by matthdsm** [comment]
Thanks for the review @drpatelh , I think everything's fixed!

**Review by drpatelh — APPROVED** [review]
🤩

**Pull request merged**